common/rpm.go

```diff
@@ -275,6 +277,16 @@ func (rpm *ringpopMonitorImpl) FindHostForKey(service string, key string) (*Host
 	return &host, nil
 }
 
+// FindRandomHostForKey finds and returns a random host responsible for handling the given key
+func (rpm *ringpopMonitorImpl) FindRandomHostForKey(service string) (*HostInfo, error) {
+	members, err := rpm.GetHosts(service)
+	if err != nil || members == nil || len(members) == 0 {
```

Comment:
GetHosts() already does all of these checks for you and returns the correct error code. Just check for err != nil here